### PR TITLE
Cavity Display: Created unit test for fault.py and updated PV object generation

### PR DESCRIPTION
**Pull request by Derikka**
In fault.py:
- Brought fault object PV storage in line with lazy generation from SC linac objects
- Changed parameters in __init__ to be `None` to make writing unit tests easier

In test_fault.py:
- Implemented most unit tests for `Fault` class
- Commented out `test_was_faulted` method because it threw an error when running all tests at once, but threw no errors when running test and debugger locally in pyCharm
- Still need to do something for `test_get_fault_count_over_time_range`. Skipped for now

### CODE REVIEW

**Comment by lisazacarias** [comment]
@Derikka can you fix the merge conflicts? I didn't realize your two requests conflicted lol

**Comment by Derikka** [comment]
Okay, fixed the merge conflicts! I think I broke things differently though. The frontend/utils.py file may be broken by the new pv_obj property added to fault.py (implemented to make it so a PV object isn't created until its first access). Needs further investigation

File: displays/cavity_display/backend/fault.py [hunk 1]
```diff
@@ -114,20 +118,17 @@ class AlarmSeverity(DefaultIntEnum):
             )
 
     def was_faulted(self, time: datetime) -> bool:
-        archiver_value: ArchiverValue = get_data_at_time(
-            # Changed from self.pv.pvname
-            pv_list=[self.pv], time_requested=time
-        )[self.pv]
+        archiver_result = get_data_at_time(pv_list=[self.pv], time_requested=time)
```
[the remark this answers is not in the document]
Comment (Derikka) [reply displays/cavity_display/backend/fault.py:121]: Ooh, yes! Adding that now

File: tests/displays/cavity_display/backend/test_fault.py [hunk 2]
```diff
@@ -58,35 +48,32 @@ def test_is_faulted_invalid(self):
...
         self.fault.fault_value = 5
         self.assertTrue(self.fault.is_faulted(pv))
 
         self.fault.fault_value = 3
         self.assertFalse(self.fault.is_faulted(pv))
 
     def test_is_faulted_exception(self):
-        pv = make_mock_pv()
-        # Set both values to None to trigger the exception
-        self.fault.ok_value = None
```
[the remark this answers is not in the document]
Comment (Derikka) [reply tests/displays/cavity_display/backend/test_fault.py:85]: I think they're no longer necessary now that we changed the init in fault.py to set those values by default to None

[the remark this answers is not in the document]
Comment (Derikka) [reply tests/displays/cavity_display/backend/test_fault.py:85]: For easier user readability though, totally fine with keeping them!

[the remark this answers is not in the document]
Comment (lisazacarias) [reply tests/displays/cavity_display/backend/test_fault.py:85]: Oh you're right! ignore me

File: tests/displays/cavity_display/backend/test_fault.py [hunk 3]
```diff
@@ -12,36 +10,28 @@
...
 
+from displays.cavity_display.backend.fault import FaultCounter
+
 archiver_value = ArchiverValue()
 get_data_at_time_mock = MagicMock(return_value={"PV": archiver_value})
 get_values_over_time_range_mock = MagicMock(
     return_value={"PV": ArchiveDataHandler([archiver_value])}
 )
 
+archiver_value = ArchiverValue()
```
[the remark this answers is not in the document]
Comment (lisazacarias) [reply tests/displays/cavity_display/backend/test_fault.py:21]: I'm not sure if lines 16-18 are duplicated too

[the remark this answers is not in the document]
Comment (Derikka) [reply tests/displays/cavity_display/backend/test_fault.py]: Okay, condensed them!

File: tests/displays/cavity_display/backend/test_fault.py [hunk 4]
```diff
@@ -12,36 +10,28 @@
...
+from displays.cavity_display.backend.fault import FaultCounter
+
 archiver_value = ArchiverValue()
 get_data_at_time_mock = MagicMock(return_value={"PV": archiver_value})
 get_values_over_time_range_mock = MagicMock(
     return_value={"PV": ArchiveDataHandler([archiver_value])}
 )
 
+with mock.patch(
+    "lcls_tools.common.data.archiver.get_data_at_time", get_data_at_time_mock
```
Comment (lisazacarias) [inline tests/displays/cavity_display/backend/test_fault.py:22]: I think this is also taken care of in line 29?

Comment (Derikka) [reply tests/displays/cavity_display/backend/test_fault.py]: You're right! I thought it had something to do with the import Fault, but I just moved that import Fault statement and deleted the duplicate code and it seems fine still!

**Review by lisazacarias — APPROVED** [review]

**Pull request merged**